### DISCUSSION TITLE
The code portion for constructing the SyntaxTree at parsing stage

### DISCUSSION THREAD

**Discussion by biwst1** [dotnet/roslyn#74633] [General]
Dear Roslyn compiler contributors,

I'm trying to trace the code at my local machine for the roslyn lexer and parser portion, in the routine "SyntaxTree ParseText()" of CSharpSyntaxTree.cs, should this routine generate the syntax tree? I see the code  "var tree = new ParsedSyntaxTree" declaring the new instance of SyntaxTree, but I could not find the routine of concrete implementation for VerifySource() in the whole roslyn repository. Could you tell me which routine(s) or class file(s) are for constructing the ParseSyntaxTree so that the represented lexical and syntactic data structure can be further fed into the Semantic Model for type info analysis. 

Thank you,

Bing Wang

**Comment by CyrusNajmabadi** [comment]
> Could you tell me which routine(s) or class file(s) are for constructing the ParseSyntaxTree 

The public entrypoint is: SyntaxFactory.ParseSyntaxTree.  However, this will only give you a syntax tree.

> can be further fed into the Semantic Model for type info analysis.

To get semantic information, you need to make a Compilation, with the right metadata references and SyntaxTrees given to it.

**Reply by 333fred** [reply]
Also, since it appears that you're talking about the roslyn testing SDK, that is here: https://github.com/dotnet/roslyn-sdk.

**Reply by biwst1** [reply]
Thank you guys, 
I found the codes to construct the "ParsedSyntaxTree": (CompilationUnitSyntax)parser.ParseCompilationUnit().CreateRed();
it returns the CSharpSyntaxNode as tree root to pass to generate a new instance of "ParsedSyntaxTree".
I'll read roslyn SDK code when I have more time.
